### PR TITLE
Solution for Shuffle Array

### PR DESCRIPTION
**Pull request by salozzo86**
Solution for [Shuffle the Array](https://leetcode.com/problems/shuffle-the-array/)

### CODE REVIEW

File: Arrays/ShuffleArray.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
+class ShuffleArray {
+    public int[] shuffle(int[] nums, int n) {
+        int length = nums.length;
+        int[] finalArray = new int[length];
+        int rightPointer = n;
```
Comment (austyhooong) [inline Arrays/ShuffleArray.java]: Great solution!

Nit: pointer could mean different thing (in C++ this means an address) so I prefer to use index

File: Arrays/ShuffleArray.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+            rightPointer++;
+            index += 2;
+        }
+        return finalArray;
+    }
+}
+
+/*
+Time Complexity: O(n)
+Space Complexity: O(n)
```
Comment (austyhooong) [inline Arrays/ShuffleArray.java]: The output of the function should not be considered towards space complexity; in this case we are not creating extra space to calculate other than the finalArray which itself is the output thus space complexity is O(1)

File: Arrays/ShuffleArray.java (hunk 1 above)
Comment (salozzo86) [reply Arrays/ShuffleArray.java]: Thanks for the advice! I've changed it to index.

File: Arrays/ShuffleArray.java (hunk 2 above)
Comment (salozzo86) [reply Arrays/ShuffleArray.java]: Very good to know! Thanks!

**Review by austyhooong — APPROVED** [review]
LGTM!

**Pull request merged**